libpod/container_internal.go

```diff
@@ -865,6 +865,12 @@ func (c *Container) init(ctx context.Context, retainRetries bool) error {
 		return err
 	}
 
+	// Setup networking prior to building container
+	if err := c.completeNetworkSetup(); err != nil {
```

Comment:
I feel like we've entirely defeated the point of `postConfigureNetNS` with this change. If we're unconditionally doing it before the container is created, there's no further point to the code - we should remove it and simplify the logic.

Reply:
Would we want runc to create a network namespace as well, or have a more appropriately named variable to differentiate that logic?

Reply:
I like the idea of simplifying and having a consistent flow.

I would weigh in more of it was clear the scenario/use-case around postConfigureNetNS - not to nit, but can someone describe the comment on that?  Happy to send a PR to add here:
https://github.com/containers/libpod/blob/master/libpod/container.go#L396-L398

Reply:
I'd make creating the networking namespace and passing it into the runtime unconditional.

Reply:
I am not sure how easily we can take rid of `postConfigureNetNS`.  It is also needed when a user namespace is created by the OCI runtime.  If we create the network namespace before the OCI runtime runs and configures the user namespace, then the network namespace is owned by the wrong user namespace